Load of helm chart catalog throws and exception if users cannot see helm repos

<!--------- For bugs and general issues --------->
**Setup**
- Rancher version: v2.8.0

**Describe the bug**
- If a user doesn't have permission to view helm repos an error is thrown
- ```
  fetch.client.js:79 Error in fetch(): TypeError: Cannot read properties of undefined (reading 'filter')
    at _callee$ (catalog.js:357:1)
    at tryCatch (regeneratorRuntime.js:44:1)
    at Generator.<anonymous> (regeneratorRuntime.js:125:1)
    at Generator.next (regeneratorRuntime.js:69:1)
    at asyncGeneratorStep (asyncToGenerator.js:3:1)
    at _next (asyncToGenerator.js:22:1)
  ```
- Regression brought in by // https://github.com/rancher/dashboard/pull/8754/files#diff-31b0043fb6852fd3650396c7d7fbe3b5ae4ba1fe3a4293fadb00a712ac53684cR357

**To Reproduce**
- Create a user with only the global `user-base` role
- Assign them as a cluster owner to an rke1 cluster
- Log in as that user
- Go to Edit the rke1 cluster

**Result**
- Exception thrown

**Expected Result**
- No exception thrown

**Screenshots**
<!-- If applicable, add screenshots to help explain your problem and/or errors in the browser's dev console -->

**Additional context**
- Given the user's roles the edit page won't display correctly, even after any fix is applied

--- a/shell/store/catalog.js
+++ b/shell/store/catalog.js
@@ -354,7 +354,7 @@ export const actions = {
 
     // As per comment above, when there are no clusters this will be management. Store it such that it can be used for those cases
     commit('setInStore', inStore);
-    hash.cluster = hash.cluster.filter((repo) => !(repo?.metadata?.annotations?.[CATALOG_ANNOTATIONS.HIDDEN_REPO] === 'true'));
+    hash.cluster = hash.cluster?.filter((repo) => !(repo?.metadata?.annotations?.[CATALOG_ANNOTATIONS.HIDDEN_REPO] === 'true'));
 
     commit('setRepos', hash);
 

